demos/gesture_recognition_demo/cpp_gapi/src/custom_kernels.cpp

```diff
@@ -0,0 +1,196 @@
+        ar_state = std::make_shared<ARStateParams>(ARStateParams{frames, 0});
+    }
+
+    static void run(const cv::Mat& in_frame,
+                    const TrackedObjects& tracked_persons,
+                    const std::shared_ptr<int>& runtime_id,
+                    const cv::Scalar& net_size,
+                          std::vector<cv::Mat> &prepared_mat,
+                    ARStateParams& ar_state) {
+        const int channels = int(net_size[0]);
+        const int batch = int(net_size[1]);
+        const int height = int(net_size[2]);
+        const int width = int(net_size[3]);
+        cv::Size sz_dst(height, width);
+        if (ar_state.stored_frames.size() < size_t(batch)) {
+            ar_state.stored_frames.emplace_back(in_frame.clone());
+        } else {
+            if(prepared_mat.size() == 0) {
+                cv::Mat mat_to_process;
+                mat_to_process.create({1, channels, batch, height, width}, CV_32F);
+                // FIXME: Why is GArray result?
+                prepared_mat.push_back(mat_to_process);
+            }
+            ar_state.stored_frames.pop_front();
+            ar_state.stored_frames.push_back(in_frame.clone());
+
+            if (size_t(*runtime_id) < tracked_persons.size()) {
+                ar_state.last_id = *runtime_id;
+            }
+            int step = 0;
+            for(const auto& frame : ar_state.stored_frames) {
+                cv::Mat crop, cvt, resized;
+                if (tracked_persons.size() > 0 &&
+                    size_t(ar_state.last_id) < tracked_persons.size()) {
+                    cv::Rect roi = tracked_persons.at(ar_state.last_id).rect;
+
+                    crop = frame(convert_to_central_roi(roi,
+                                                        cv::Size(height, width),
+                                                        ACTION_IMAGE_SCALE));
+                } else {
+                    crop = frame;
+                }
+                crop.convertTo(cvt, CV_32F);
+                cv::resize(cvt, resized, sz_dst);
+
+                cv::Mat different_channels[3];
+                cv::split(resized, different_channels);
+                cv::Mat matB = different_channels[0];
+                cv::Mat matG = different_channels[1];
+                cv::Mat matR = different_channels[2];
+
+                std::copy_n(matR.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 0 * batch * height * width
+                                + step * height * width);
+                std::copy_n(matG.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 1 * batch * height * width
+                                + step * height * width);
+                std::copy_n(matB.begin<float>(), height * width,
+                            prepared_mat[0].begin<float>() + 2 * batch * height * width
+                                + step * height * width);
```

Comment:
What I'd expect here is
1. The preprocessing part (`crop`, `resize`, `covertTo`, `split`) - happening only once per a new frame - if it is possble.
2. **Only** copy_n happenning for all (past+new) preprocessed frames to form the resulting blob.